src/components/legend/draw.js

```diff
@@ -319,15 +319,17 @@ function _draw(gd, legendObj) {
                         var transform = Drawing.getTranslate(legend);
                         x0 = transform.x;
                         y0 = transform.y;
+                        var bbox = legend.node().getBBox();
```

Comment:
Thanks for the PR.
It appears [`getBBox`](https://developer.mozilla.org/en-US/docs/Web/API/SVGGraphicsElement/getBBox) is not used in our source code. 
Could you possibly use [`getBoundingClientRect`](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect) instead?

On another note: shouldn't these value only be used when `layout.legend.xanchor !== 'left'` or `layout.legend.yanchor !== 'top'`? i.e. to keep the current behavior of the cases that work fine at the moment.

Reply:
This PR doesn't change the current behavior when `xanchor === 'left`' or `yanchor === 'right'`, it just makes dragging work for any combination of xanchor and yanchor.

I'm happy to use `getBoundingClientRect` instead, though looking into it more deeply I wonder if we should be using `legendObj._width` and `legendObj._height`; those values seem to be used to do the positioning in the first place: https://github.com/plotly/plotly.js/blob/master/src/components/legend/draw.js#L137

I tried this change, and it seems to work great:
```diff
diff --git a/src/components/legend/draw.js b/src/components/legend/draw.js
index f63934d86..d27ce2d5f 100644
--- a/src/components/legend/draw.js
+++ b/src/components/legend/draw.js
@@ -319,9 +319,8 @@ function _draw(gd, legendObj) {
                         var transform = Drawing.getTranslate(legend);
                         x0 = transform.x;
                         y0 = transform.y;
-                        var bbox = legend.node().getBBox();
-                        legendWidth = bbox.width;
-                        legendHeight = bbox.height;
+                        legendWidth = legendObj._width;
+                        legendHeight = legendObj._height;
                     },
                     moveFn: function(dx, dy) {
                         var newX = x0 + dx;
```

Reply:
Great. Let's use `._width` & `._height` :)

Reply:
Ok, I made the change.